python/modules/IcePy/Current.cpp

```diff
@@ -308,12 +308,20 @@ PyObject*
 IcePy::createCurrent(const Ice::Current& current)
 {
     //
-    // Return an instance of IcePy.Current to hold the current information.
+    // Return an instance of Ice.Current to hold the current information.
```

Comment:
Ice.Current is an abstract class defined in Python and implemented by the private class CurrentI. The CurrentI class acts as a wrapper around IcePy.Current. This setup, while somewhat involved, is designed to avoid exposing IcePy types directly in our public API.

Reply:
It sounds very complicated for something very simple such as Current.

Why is Current abstract in the first space? We can't just use a concrete Ice.Current fully defined in Python?

Reply:
The `IcePy.Current` class holds a pointer to the C++ current object, and its getter lazily creates the Python objects for each field. This optimization avoids creating fields that might never be accessed by the dispatch. However, defining a concrete `Current` class in Python and fully initializing it from the start would be simpler.

Reply:
Assuming we can reuse the same ObjectAdapter and Connection object over and over (not create new one for each dispatch), I'd create a populated Python Ice.Current, even if the application doesn't use it (which is very common).

Reply:
Fixed in #2552